docs/0012-hash-and-no-prefix-id-n-tuple-storage-layout.md

```diff
@@ -0,0 +1,429 @@
+MUST also equal `0`.
+
+The product of `numberOfTuples` and `tupleSize` MUST be less than or equal to
+the number of characters in the hex encoded digest.
+
+#### delimiters
+
+`delimiters` is a list of case-sensitive delimiters that will be used to strip the prefix from the
+OCFL object identifier. 
+
+If it is empty, this extension is equivalent to `0003-hash-and-id-n-tuple-storage-layout` as no prefix will be removed.
+
+If a delimiter occurs at the end of the OCFL object identifier, that occurrence of a delimiter is ignored and the last previous one, if present, is used.
+
+If an array of delimiters is used, there is no precedence of the delimiters; however, the final occurrence of any delimiter in the array identifies the end of the prefix.
+
+| Object ID | Delimiter | Object id with prefix removed |
+| --- | --- | --- |
+| abcd | ["d"] | abcd |
+| abcd | ["c", "d"] | d |
+| abcdd | ["d"] | d |
+
+## Procedure
+
+The following is an outline of the steps to follow to map an OCFL object
+identifier to an OCFL object root path using this extension (also see the
+["Python Code"](#python-code) section):
+
+1. Remove the prefix, which is everything to the left of the right-most instance of any of the delimiters, as well as the corresponding delimiter. If there is no delimiter, the whole id is used; if a delimiter is found at the end, it is ignored.
+2. The OCFL object identifier with prefix removed is encoded as UTF-8 and hashed using the specified
+   `digestAlgorithm`.
+3. The digest is encoded as a lower-case hex string.
+4. Starting at the beginning of the digest and working forwards, the digest is
+   divided into `numberOfTuples` tuples each containing `tupleSize` characters.
+5. The tuples are joined, in order, using the filesystem path separator.
+6. The OCFL object identifier with prefix removed is percent-encoded to create the encapsulation
+   directory name (see ["Encapsulation Directory"](#encapsulation-directory)
+   section above for details).
+7. The encapsulation directory name is joined to the end of the path.
+
+## Examples
+
+### Example 1
+
+This example demonstrates what the OCFL storage hierarchy looks like when using
+this extension's default configuration.
+
+#### Parameters
+
+It is not necessary to specify any parameters to use the default configuration.
+However, if you were to do so, it would look like the following:
+
+```json
+{
+    "extensionName": "0012-hash-and-no-prefix-id-n-tuple-storage-layout",
+    "digestAlgorithm": "sha256",
+    "tupleSize": 3,
+    "numberOfTuples": 3,
+    "delimiters": []
+}
+```
+
+#### Mappings
+
+| Object ID | Object ID with prefix removed | Digest | Object Root Path |
+| --- | --- | --- | --- |
+| `object-01` | `object-01` | `3c0ff4240c1e116dba14c7627f2319b58aa3d77606d0d90dfc6161608ac987d4` | `3c0/ff4/240/object-01` |
+| `..hor/rib:le-$id` | `rib:le-$id` | `a6b979238e7131d89e45b913942c33374ce5c09d348727b50c5f995d0ce4f7f8` | `a6b/979/238/rib%3ale-%24id` |
```

Comment:
This example mapping (`..hor/rib:le-$id` --> `rib:le-$id`) is no longer now that you are using the default "delimiters" parameter.

Reply:
thanks! done